python/ray/serve/tests/test_logging.py

```diff
@@ -280,6 +286,172 @@ def check_log_file(log_file: str, expected_regex: list):
+                return {
+                    "logs_path": logger.handlers[1].baseFilename,
+                }
+
+        serve.run(Model.bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+
+        paths = resp["logs_path"].split("/")
+        paths[-1] = "new_dir"
+        new_log_dir = "/".join(paths)
+
+        serve.run(Model.options(logging_config={"logs_dir": new_log_dir}).bind())
+        resp = requests.get("http://127.0.0.1:8000/").json()
+        assert "new_dir" in resp["logs_path"]
+
+        check_log_file(resp["logs_path"], [".*model_info_level.*"])
+
+    @pytest.mark.parametrize("enable_access_type", [True, False])
+    def test_access_log(self, serve_and_ray_shutdown, enable_access_type):
+        logger = logging.getLogger("ray.serve")
+
+        @serve.deployment(logging_config={"enable_access_log": enable_access_type})
+        class Model:
+            def __call__(self, req: starlette.requests.Request):
+                if enable_access_type:
+                    assert len(logger.handlers) == 2
+                else:
+                    assert len(logger.handlers) == 1
+                return
+
+        serve.run(Model.bind())
+
+        resp = requests.get("http://127.0.0.1:8000/")
+        assert resp.status_code == 200
+
+    def test_application_logging_overwrite(self, serve_and_ray_shutdown):
+        @serve.deployment
+        class Model:
+            def __call__(self, req: starlette.requests.Request):
+                logger = logging.getLogger("ray.serve")
+                logger.info("model_info_level")
+                logger.debug("model_debug_level")
+                return {
+                    "log_file": logger.handlers[1].baseFilename,
+                }
+
+        serve.run(Model.bind(), logging_config={"log_level": "DEBUG"})
+        resp = requests.get("http://127.0.0.1:8000/").json()
+        expected_log_regex = [".*model_info_level.*", ".*model_debug_level.*"]
+        check_log_file(resp["log_file"], expected_log_regex)
+
+        # Setting logging config in the deployment level, application
+        # config can't override it.
+
+        @serve.deployment(logging_config={"log_level": "INFO"})
+        class Model2:
+            def __call__(self, req: starlette.requests.Request):
+                logger = logging.getLogger("ray.serve")
+                logger.info("model_info_level")
+                logger.debug("model_debug_level")
+                return {
+                    "log_file": logger.handlers[1].baseFilename,
+                }
+
+        serve.run(
+            Model2.bind(),
+            logging_config={"log_level": "DEBUG"},
+            name="app2",
+            route_prefix="/app2",
+        )
+        resp = requests.get("http://127.0.0.1:8000/app2").json()
+        # Make sure 'model_debug_level' log content does not exist
+        with pytest.raises(AssertionError):
+            check_log_file(resp["log_file"], [".*model_debug_level.*"])
```

Comment:
Should also assert that the INFO level logs exist, no?

Reply:
done.